### ISSUE TITLE
DecimalPipe incorrectly rounds negative numbers when the fraction is .5

### ISSUE THREAD

**Issue by shahmirn** [angular/angular#24038]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

{{-2.5 | number:'1.0-0'}} displays -3 in the template

## Expected behavior

{{-2.5 | number:'1.0-0'}} should follow the same rules as Math.round(-2.5) and display -2 in the template

## Minimal reproduction of the problem with instructions

https://stackblitz.com/edit/angular-gitter-msxufr

## Further details
The issue stems from formatNumber, which calls formatNumberToLocaleString, which calls parseNumber, which does Math.abs at https://github.com/angular/angular/blob/master/packages/common/src/i18n/format_number.ts#L316

Afterwards, formatNumberToLocaleString at the end at https://github.com/angular/angular/blob/master/packages/common/src/i18n/format_number.ts#L116 re-adds the negative symbol.

## Environment

<pre><code>
Angular version: X.Y.Z

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by s-gbz** [comment]
Hello, this seems like a reasonable issue for a first time contributor, hence I would love to be assigned for that :)

**Comment by vicb** [comment]
I agree it's a bit awkward but which behavior is correct is not simple, see the different types of rounding at https://en.wikibooks.org/wiki/Arithmetic/Rounding.

While behaving the same as JS would probably be better, it would also be a breaking change for people relying on the current behavior.

May be one way to fix would be to:
- Document the current behavior in the API docs,
- Add an optional parameter to the API and pipe for the kind of rounding we want that would default to `AWAY_FROM_ZERO`, i.e. the current behavior.

What do you guys think ?

**Comment by shahmirn** [comment]
@vicb That sounds like a good solution to ensure that it's a non-breaking change and would work for me.

**Comment by Miced** [comment]
It would be nice to mention the rounding in the docs, and perhaps provide some extra options (flooring, ceiling etc). Currently there is no mention that the number is being rounded.

**Comment by vicb** [comment]
Would somebody here be willing to submit PRs to 
- update the current API docs,
- being able to configure the kind of rounding.

**Comment by Miced** [comment]
I can prepare a PR if we could agree on the notation. I checked the docs and noticed that `[formatNumberToLocaleString()](https://github.com/angular/angular/blob/master/packages/common/src/i18n/format_number.ts#L24-L123)` is used in percent and currency pipes as well. It is possible to extend the behaviour to those pipes as well, or keep the same default behaviour.

**Comment by vicb** [comment]
I probably makes sense to extend that to all pipes. 
May be you can do a short proposal so that we agree before jumping to the impl.
Also could someone take care of the API docs (describing the current behavior).
Thanks !

**Comment by s-gbz** [comment]
I will take care of it Monday or Tuesday.

**Comment by Miced** [comment]
I found some time to look into this again, and it seems that no breaking change would occur if we used the same changes on the rest pipes. I have 2 possible solutions for this proposal:

- **Enhancing the format string.** I think that this is the most compact solution since it does not create extra pipe parameters. Using the current format string, it could be extended as: `{minIntegerDigits}.{minFractionDigits}-{maxFractionDigits}*{rounding}`. Using the `*` symbol as delimiter seems correct to me in order to differentiate from the previous `-` and is single character notation, although brackets might resemble more like rounding (`{minIntegerDigits}.{minFractionDigits}-{maxFractionDigits}[{rounding}]`).
- **Adding a new parameter as last.** This might be more clean but  given that e.g. currency pipe has already 4 parameters and locale is already used as last parameter in all pipes, it is possible that the developer does not want to override the locale parameter just to get to the rounding parameter.
 
In both cases the proposed possible parameter values would comply to the aforementioned [wiki](https://en.wikibooks.org/wiki/Arithmetic/Rounding) and would be:
  - `ceiling`
  - `floor`
  - `away_from_zero`
  - `to_zero`
  - `to_nearest`: The current behavior
